src/dune_pkg/opam_solver.ml

```diff
@@ -1452,8 +1452,33 @@ let opam_variable_to_slang ~loc packages variable =
             Blang.Expr (convert_with_package_name package_name))))
 ;;
 
+(* Handles the special case for packages whose names contain '+' characters
+   where a special form of string interpolation is used. From the opam manual:
+   Warning: if the package name contains a + character (e.g. conf-g++), their
+   variables may only be accessed using opam 2.2 via string interpolation,
+   with the following syntax:
+
+     "%{?conf-g++:your-variable:}%"
+*)
+let desugar_special_string_interpolation_syntax
+      ((packages, variable, string_converter) as fident)
+  =
+  match string_converter with
+  | Some (package_and_variable, "")
+    when List.is_empty packages && OpamVariable.to_string variable |> String.is_empty ->
```

Comment:
```suggestion
    when List.is_empty packages && String.is_empty (OpamVariable.to_string variable) ->
```
Minor suggestion, but it found is pretty hard to parse the function in this context.

Reply:
Done!